nix/pkgs.nix

```diff
@@ -25,9 +25,18 @@ let
 
   # Override some packages and utilities
   pkgsOverlay = import ./overlay.nix;
+
+  # Overriding NIXPKGS system: x86_64-darwin (Apple Silicon fix for android-sdk, see PR-16237)
+  system = if builtins.getEnv "CI" == "true" then
+    builtins.currentSystem
+  else if builtins.currentSystem == "aarch64-darwin" then
+    builtins.trace "Overriding NIXPKGS system: x86_64-darwin (Apple Silicon fix for android-sdk, see PR-16237)" "x86_64-darwin"
+  else
+    builtins.currentSystem;
```

Comment:
This could be made simpler:
```nix
{
  # Fix for lack of Android SDK for M1 Macs.
  systemOverride = let
    inherit (builtins) currentSystem getEnv trace;
    systemOverrideMsg = "Overriding NIXPKGS system: x86_64-darwin (Apple Silicon fix for android-sdk)";
  in
    if getEnv "CI" == "" && currentSystem == "aarch64-darwin" then
      trace systemOverrideMsg "x86_64-darwin"
    else
      currentSystem;
}
```
And use `system = systemOverride` later on for clarity.
Not sure if we need that warning message. It will show up twice every time due to use of `gcroots.sh`.

Reply:
Done, applied in the latest commit.